src/commonMain/kotlin/org/kosat/heuristics/VariableSelector.kt

```diff
@@ -0,0 +1,67 @@
+package org.kosat.heuristics
+
+import org.kosat.CDCL
+import org.kosat.Clause
+import org.kosat.Lit
+import kotlin.math.abs
+
+abstract class VariableSelector {
+    protected var assumptions: List<Lit> = emptyList()
+
+    fun initAssumptions(assumptions: List<Lit>) {
+        this.assumptions = assumptions
+    }
+
+    abstract fun build(clauses: List<Clause>)
+    abstract fun nextDecisionVariable(vars: List<CDCL.VarState>, level: Int): Int
+    abstract fun addVariable()
+    abstract fun update(lemma: Clause)
+}
+
+class VSIDS(private var varsNumber: Int = 0) : VariableSelector() {
+    private val decay = 50
+    private val divisionCoeff = 2.0
+    private var numberOfConflicts = 0
+
+    override fun update(lemma: Clause) {
+        lemma.forEach { lit -> activity[abs(lit)]++ } //todo litIndex
+        numberOfConflicts++
+        if (numberOfConflicts == decay) {
+            // update scores
+            numberOfConflicts = 0
+            activity.forEachIndexed { ind, _ -> activity[ind] /= divisionCoeff }
+        }
+    }
+
+    override fun addVariable() {
+        activity.add(0.0)
+        varsNumber++
+    }
+
+    private val activity = mutableListOf<Double>()
+    override fun build(clauses: List<Clause>) {
+        while (activity.size < varsNumber + 1) {
+            activity.add(0.0)
+        }
+        clauses.forEach { clause ->
+            clause.forEach { lit ->
+                activity[abs(lit)] += 1.0
+            }
+        }
+    }
+
+    override fun nextDecisionVariable(vars: List<CDCL.VarState>, level: Int): Int {
+        return if (level > assumptions.size) {
+            vsids(vars)
+        } else {
+            assumptions[level - 1]
+        }
+    }
+
+    // Looks for index of undefined variable with max activity
+    private fun vsids(vars: List<CDCL.VarState>) = (1..varsNumber).filter {
```

Comment:
Method name `vsids` does not describe what this method actually does.
Also, it is better to specify the return type, when the function is not one-liner, or when this type is not obvious, or when this type would not help understand the method. In your case, the contrary to all points holds: the method is not one-liner, the return type could help understand what `vsids` method actually does, and the type is actually not obvious (you start with `List<Int>`, then convert it to `Int` in `let`-block — all of this requires actually reading the code in order to guess the type.
According to the docs (at last, there is a documentation!), the good name would be `getMaxActivityVariable(vars: List<VarState>) -> Var`, where `Var` is whatever type (typedef, inline class, or just `Int`) you are using for variables. Btw, you could import `...CDCL.VarState` in order to use `VarState` without `CDCL.` prefix.